### PR TITLE
Minor cleanup in scheduler/PriorityQueue

### PR DESCRIPTION
**Pull request by misterikkit**
**What type of PR is this?**
/kind cleanup
/sig scheduling

**What this PR does / why we need it**:
This fixes a spurious error that was being returned by PriorityQueue and adds some early returns to make the code easier to read.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 1]
```diff
@@ -276,32 +276,37 @@ func (p *PriorityQueue) run() {
...
+	for i, np := range p.nominatedPods[nnn] {
+		if np.UID != pod.UID {
+			continue
 		}
+		p.nominatedPods[nnn] = append(p.nominatedPods[nnn][:i], p.nominatedPods[nnn][i+1:]...)
+		if len(p.nominatedPods[nnn]) == 0 {
+			delete(p.nominatedPods, nnn)
+		}
+		break
+
```
Comment (bsalamat) [inline pkg/scheduler/internal/queue/scheduling_queue.go]: nit: please remove the blank line.

**Review by bsalamat — COMMENTED** [review]
just a minor nit. Otherwise, LGTM.

File: pkg/scheduler/internal/queue/scheduling_queue.go (hunk 1 above)
Comment (misterikkit) [reply pkg/scheduler/internal/queue/scheduling_queue.go]: done

**Review by bsalamat — APPROVED** [review]
/lgtm
/approve

**Pull request merged**